src/InlineRelationsTransformer.cpp

```diff
@@ -629,9 +629,23 @@ NullableVector<AstArgument*> getInlinedArgument(AstProgram& program, const AstAr
                     break;
                 }
             }
-        } else if (dynamic_cast<const AstUserDefinedFunctor*>(arg)) {
-            // TODO (azreika): extend to handle user-defined functors
-            assert(false && "unhandled argument: AstUserDefinedFunctor");
+        } else if (const auto* udf = dynamic_cast<const AstUserDefinedFunctor*>(arg)) {
+            for (size_t i = 0; i < udf->getArgCount(); i++) {
+                // try inlining each argument from left to right
+                NullableVector<AstArgument*> argumentVersions =
+                        getInlinedArgument(program, udf->getArg(i));
+                if (argumentVersions.isValid()) {
+                    changed = true;
+                    for (AstArgument* newArgVersion : argumentVersions.getVector()) {
+                        // same functor but with new argument version
+                        AstUserDefinedFunctor* newFunctor = udf->clone();
+                        newFunctor->setArg(i, std::unique_ptr<AstArgument>(newArgVersion));
```

Comment:
Is this safe? You seem to be making a unique pointer out of the contents of argumentVersions, but it's not clear to me if everything allocated gets deleted, and deleted only once.

Reply:
The inlining code should be properly refactored in a future PR to fix this problem, but this pattern is used throughout that transformation whenever `getInlinedArgument` is called, which should be fine in this case as it will always return a vector of clones/new objects.